javascript/javascript1/week4/voice-assistant.js

```diff
@@ -0,0 +1,44 @@
+let userName = [];
+let myTodo = [];
+function getReply(command) {
+  if (command.includes("my name is")) {
+    let commandStr = command.split(" ");
+    let activeName = commandStr[commandStr.length - 1];
+
+    if (userName.includes(activeName)) {
+      console.log(" Yes I know");
+      return;
+    }
+
+    userName.push(activeName);
+    console.log(`nice to meet you ${activeName}`);
+    return;
+  }
+
+  if (command.includes("what is my name")) {
+    if (!userName) {
+      console.log("what is your name");
+      return;
+    }
+    console.log(`your name is ${activeName}`);
```

Comment:
`activeName` isn't available in this scope. 
You can do something like `userName[userName.length - 1]` to get the last name in the array.